src/components/CursorInfo/CursorInfoComponent.tsx

```diff
@@ -0,0 +1,139 @@
+                zCoordString.push(frame.spectralInfo.velocityString.replace(/\w+:\s/, "\u000A"));
+            }
+        } else {
+            zCoordString.push("NaN");
+        }
+        return zCoordString;
+    };
+
+    render() {
+        const appStore = AppStore.Instance;
+        const frameNum = appStore.frames.length;
+        const frame = appStore.activeFrame;
+
+        if (frameNum <= 0) {
+            return (
+                <div className="region-list-widget">
+                    <NonIdealState icon={"folder-open"} title={"No file loaded"} description={"Load a file using the menu"} />
+                </div>
+            );
+        }
+
+        const columnWidths = [90, 95, 50, 95, 95, 128, 70, 70];
+        const columnNames = ["Image", "Value", "WCS", "XY (World)", "XY (Image)", "Z", "Channel", "Stokes"];
+        const dataType = CARTA.ColumnType.String;
+        const columnHeaders = columnNames.map((name, index) => new CARTA.CatalogHeader({name: name, dataType, columnIndex: index}));
+
+        const imageNames = appStore.frames.map(frame => frame.filename);
+        let values = Array(frameNum).fill("-");
+        let systems = Array(frameNum).fill("-");
+        let worldCoords = Array(frameNum).fill("-");
+        let imageCoords = Array(frameNum).fill("-");
+        const zCoords = appStore.frames.map(frame => this.genZCoordString(frame));
+        const channels = appStore.frames.map(frame => frame.requiredChannel);
+        const stokes = appStore.frames.map(frame => frame.requiredStokes);
+
+        const activeFrameIndex = appStore.activeFrameIndex;
+        if (frame.cursorInfo.isInsideImage && frame.cursorValue !== undefined) {
+            let valueString = formattedExponential(frame.cursorValue.value, 5, "", true, true);
+            if (isNaN(frame.cursorValue.value)) {
+                valueString = "NaN";
+            }
+            if (!frame.isCursorValueCurrent) {
+                valueString += "*";
+            } else {
+                valueString += " ";
+            }
+            values[activeFrameIndex] = (
+                <React.Fragment>
+                    {valueString}
+                    <br />
+                    {frame.unit}
+                </React.Fragment>
+            );
+        }
+        systems[activeFrameIndex] = appStore.overlayStore.global.explicitSystem;
+        worldCoords[activeFrameIndex] = (
+            <React.Fragment>
+                {frame.cursorInfo?.infoWCS?.x}
+                <br />
+                {frame.cursorInfo?.infoWCS?.y}
+            </React.Fragment>
+        );
+        imageCoords[activeFrameIndex] = (
+            <React.Fragment>
+                {toFixed(frame.cursorInfo?.posImageSpace?.x, 3)}
+                <br />
+                {toFixed(frame.cursorInfo?.posImageSpace?.y, 3)}
+            </React.Fragment>
+        );
+
+        let columnsData = new Map<number, any>();
+        columnsData.set(0, {dataType, data: imageNames});
```

Comment:
elsewhere in the codebase we use the `Map` array constructor (e.g. `new Map([[key1, val1], [key2, val2]])`. I'm not sure if that would be neater or not 🤔

Reply:
Yes please 😊

Reply:
Changed both parts